# HG changeset patch
# User Jacob Dawid <jacob.dawid@gmail.com>
# Date 1343634957 -7200
#      Mon Jul 30 09:55:57 2012 +0200
# Branch gui
# Node ID 6889217b9d786c5347ad55e9a76bbb6b629fd950
# Parent  ab771158224ef3de27541319c4a2bed75cc5b871
Modified GUI sources so they can be build inside the source tree, without installing octave first.

* octave-event.h: Modified includes.
* octave-link.h: Modified includes.
* src.pro: Removed all mkoctfile references and added paths.
* symbol-information.h: Modified includes.

diff --git a/gui/src/octave-adapter/octave-event.h b/gui/src/octave-adapter/octave-event.h
--- a/gui/src/octave-adapter/octave-event.h
+++ b/gui/src/octave-adapter/octave-event.h
@@ -15,17 +15,17 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
 #ifndef OCTAVEEVENT_H
 #define OCTAVEEVENT_H
 
 #include <string>
 #include "octave-event-observer.h"
-#include "octave/config.h"
+#include "config.h"
 #include "symtab.h"
 #include "oct-env.h"
 #include "pt-eval.h"
 #include "toplev.h"
 #include "parse.h"
 #include "debug.h"
 #include "cmd-hist.h"
 
diff --git a/gui/src/octave-adapter/octave-link.h b/gui/src/octave-adapter/octave-link.h
--- a/gui/src/octave-adapter/octave-link.h
+++ b/gui/src/octave-adapter/octave-link.h
@@ -20,44 +20,44 @@
 
 // Octave includes
 #undef PACKAGE_BUGREPORT
 #undef PACKAGE_NAME
 #undef PACKAGE_STRING
 #undef PACKAGE_TARNAME
 #undef PACKAGE_VERSION
 #undef PACKAGE_URL
-#include "octave/config.h"
-#include "octave/cmd-edit.h"
-#include "octave/error.h"
-#include "octave/file-io.h"
-#include "octave/input.h"
-#include "octave/lex.h"
-#include "octave/load-path.h"
-#include "octave/octave.h"
-#include "octave/oct-hist.h"
-#include "octave/oct-map.h"
-#include "octave/oct-obj.h"
-#include "octave/ops.h"
-#include "octave/ov.h"
-#include "octave/ov-usr-fcn.h"
-#include "octave/symtab.h"
-#include "octave/pt.h"
-#include "octave/pt-eval.h"
-#include "octave/config.h"
-#include "octave/Range.h"
-#include "octave/toplev.h"
-#include "octave/procstream.h"
-#include "octave/sighandlers.h"
-#include "octave/debug.h"
-#include "octave/sysdep.h"
-#include "octave/ov.h"
-#include "octave/unwind-prot.h"
-#include "octave/utils.h"
-#include "octave/variables.h"
+#include "config.h"
+#include "cmd-edit.h"
+#include "error.h"
+#include "file-io.h"
+#include "input.h"
+#include "lex.h"
+#include "load-path.h"
+#include "octave.h"
+#include "oct-hist.h"
+#include "oct-map.h"
+#include "oct-obj.h"
+#include "ops.h"
+#include "ov.h"
+#include "ov-usr-fcn.h"
+#include "symtab.h"
+#include "pt.h"
+#include "pt-eval.h"
+#include "config.h"
+#include "Range.h"
+#include "toplev.h"
+#include "procstream.h"
+#include "sighandlers.h"
+#include "debug.h"
+#include "sysdep.h"
+#include "ov.h"
+#include "unwind-prot.h"
+#include "utils.h"
+#include "variables.h"
 #include "oct-mutex.h"
 
 // Standard includes
 #include <iostream>
 #include <string>
 #include <vector>
 #include <readline/readline.h>
 #include <queue>
diff --git a/gui/src/src.pro b/gui/src/src.pro
--- a/gui/src/src.pro
+++ b/gui/src/src.pro
@@ -26,36 +26,40 @@ TRANSLATIONS        += languages/generic
                        languages/pt-br.ts \
                        languages/es-es.ts \
                        languages/ru-ru.ts \
                        languages/uk-ua.ts           # Available translations
 
 win32-msvc*: include(msvc.pri)
 
 LIBS                += -lreadline -lqscintilla2  \
-                       -L../qterminal/libqterminal/$$LIBDIR_SUFFIX -lqterminal \
-                        $$system(mkoctfile -p LIBS) \
-                        $$system(mkoctfile -p OCTAVE_LIBS)
+                       -L../qterminal/libqterminal/$$LIBDIR_SUFFIX -lqterminal -lm \
+                       -L../../libcruft/.libs -lcruft \
+                       -L../../liboctave/.libs -loctave \
+                       -L../../src/.libs -loctinterp
 
 mac {
     CONFIG -= app_bundle
 }
 
 # Includepaths and libraries to link against:
-INCLUDEPATH         += . octave-adapter m-editor ../qterminal/libqterminal /usr/include/qt4 \
-                       $$system(mkoctfile -p INCFLAGS)
-INCFLAGS            += $$system(mkoctfile -p INCFLAGS)
-mac {
-    INCFLAGS += -I/opt/local/include
-}
+INCLUDEPATH         += . \
+                       octave-adapter \
+                       m-editor \
+                       ../qterminal/libqterminal \
+                       /usr/include/qt4 \
+                       ../.. \
+                       ../../src \
+                       ../../liboctave \
+                       ../../libcruft/misc
 
-QMAKE_LIBDIR        += $$system(octave-config -p OCTLIBDIR)
-unix {
-    QMAKE_RPATHDIR += $$system(octave-config -p OCTLIBDIR)
-}
+#QMAKE_LIBDIR        += $$system(octave-config -p OCTLIBDIR)
+#unix {
+#    QMAKE_RPATHDIR += $$system(octave-config -p OCTLIBDIR)
+#}
 
 mac {
     LFLAGS += -L/opt/local/lib
 }
 
 unix {
     LIBS += -lutil
 }
@@ -64,17 +68,17 @@ win32-g++ {
     QMAKE_LFLAGS += --enable-auto-import
 }
 
 win32-msvc* {
     DEFINES += QSCINTILLA_DLL
     QMAKE_CXXFLAGS += -wd4244
 }
 
-QMAKE_LFLAGS        += $$LFLAGS $$system(mkoctfile -p RLD_FLAG)
+QMAKE_LFLAGS        += $$LFLAGS
 QMAKE_CXXFLAGS      += $$INCFLAGS
 
 # Files associated with the project:
 SOURCES +=\
     octave-adapter/octave-link.cc \
     octave-adapter/octave-main-thread.cc \
     m-editor/lexer-octave-gui.cc \
     m-editor/file-editor.cc \
diff --git a/gui/src/symbol-information.h b/gui/src/symbol-information.h
--- a/gui/src/symbol-information.h
+++ b/gui/src/symbol-information.h
@@ -25,44 +25,44 @@
 
 // Octave includes
 #undef PACKAGE_BUGREPORT
 #undef PACKAGE_NAME
 #undef PACKAGE_STRING
 #undef PACKAGE_TARNAME
 #undef PACKAGE_VERSION
 #undef PACKAGE_URL
-#include "octave/config.h"
-#include "octave/cmd-edit.h"
-#include "octave/error.h"
-#include "octave/file-io.h"
-#include "octave/input.h"
-#include "octave/lex.h"
-#include "octave/load-path.h"
-#include "octave/octave.h"
-#include "octave/oct-hist.h"
-#include "octave/oct-map.h"
-#include "octave/oct-obj.h"
-#include "octave/ops.h"
-#include "octave/ov.h"
-#include "octave/ov-usr-fcn.h"
-#include "octave/symtab.h"
-#include "octave/pt.h"
-#include "octave/pt-eval.h"
-#include "octave/config.h"
-#include "octave/Range.h"
-#include "octave/toplev.h"
-#include "octave/procstream.h"
-#include "octave/sighandlers.h"
-#include "octave/debug.h"
-#include "octave/sysdep.h"
-#include "octave/ov.h"
-#include "octave/unwind-prot.h"
-#include "octave/utils.h"
-#include "octave/variables.h"
+#include "config.h"
+#include "cmd-edit.h"
+#include "error.h"
+#include "file-io.h"
+#include "input.h"
+#include "lex.h"
+#include "load-path.h"
+#include "octave.h"
+#include "oct-hist.h"
+#include "oct-map.h"
+#include "oct-obj.h"
+#include "ops.h"
+#include "ov.h"
+#include "ov-usr-fcn.h"
+#include "symtab.h"
+#include "pt.h"
+#include "pt-eval.h"
+#include "config.h"
+#include "Range.h"
+#include "toplev.h"
+#include "procstream.h"
+#include "sighandlers.h"
+#include "debug.h"
+#include "sysdep.h"
+#include "ov.h"
+#include "unwind-prot.h"
+#include "utils.h"
+#include "variables.h"
 
 /**
   * \struct symbol_information
   * \brief Meta-information over a symbol-table entry.
   * \author Jacob Dawid
   * This struct is used to store meta information over a symbol entry.
   * It reduces memory consumption, since it only stores relevant data
   * about a symbol-table entry that will be used in the model for the
